src/tsutil/Regex.cc

```diff
@@ -172,7 +172,14 @@ RegexMatches::RegexMatches(uint32_t size)
   pcre2_general_context *ctx = pcre2_general_context_create(
     &RegexMatches::malloc, [](void *, void *) -> void {}, static_cast<void *>(this));
 
-  _MatchData::set(_match_data, pcre2_match_data_create(size, ctx));
+  pcre2_match_data *match_data = pcre2_match_data_create(size, ctx);
+  if (match_data == nullptr) {
+    // buffer was too small, allocate from heap
+    debug_assert("Buffer too small, allocating from heap");
```

Comment:
You need a first parameter here for this to abort, even if it's just `false`.

Reply:
yeah, don't know what I was thinking - fixed it